chore(deps): bump golang.org/x/net from 0.33.0 to 0.38.0 in /components/{odh-}notebook-controller

jiridanek opened the pull request:
* https://github.com/opendatahub-io/kubeflow/pull/554
* https://github.com/opendatahub-io/kubeflow/pull/555

* https://github.com/opendatahub-io/kubeflow/pull/589
* https://github.com/opendatahub-io/kubeflow/pull/590

Red Hat finally shipped a 1.23 Go Toolset image, so we can get this in

* https://catalog.redhat.com/software/containers/ubi8/go-toolset/5ce8713aac3db925c03774d1

## Description

* https://github.com/opendatahub-io/kubeflow/security/dependabot/1176
* https://pkg.go.dev/vuln/GO-2025-3503

* govulncheck https://github.com/opendatahub-io/kubeflow/actions/runs/13830022809

```
Vulnerability #1: GO-2025-3503
    HTTP Proxy bypass using IPv6 Zone IDs in golang.org/x/net
  More info: https://pkg.go.dev/vuln/GO-2025-3503
  Module: golang.org/x/net
    Found in: golang.org/x/net@v0.33.0
    Fixed in: golang.org/x/net@v0.36.0

Vulnerability #2: GO-2025-34[88](https://github.com/opendatahub-io/kubeflow/actions/runs/13830022809/job/38692035697#step:4:89)
    Unexpected memory consumption during token parsing in golang.org/x/oauth2
  More info: https://pkg.go.dev/vuln/GO-2025-3488
  Module: golang.org/x/oauth2
    Found in: golang.org/x/oauth2@v0.12.0
    Fixed in: golang.org/x/oauth2@v0.27.0

Your code is affected by 0 vulnerabilities.
This scan also found 0 vulnerabilities in packages you import and 2
vulnerabilities in modules you require, but your code doesn't appear to call
these vulnerabilities.
```

## How Has This Been Tested?

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

jiridanek commented on components/notebook-controller/go.mod:
```diff
@@ -1,8 +1,7 @@
 go 1.21
-
-toolchain go1.21.9
+toolchain go1.23.7
```
I have no idea what this will do. Looking forward to see the CI on it.

jiridanek replied:
```
[1/2] STEP 11/11: RUN if [ -z ${CACHITO_ENV_FILE} ]; then go mod download; else source ${CACHITO_ENV_FILE}; fi &&   CGO_ENABLED=1 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -tags strictfipsruntime -a -o ./bin/manager main.go
go: golang.org/x/net@v0.36.0: module golang.org/x/net@v0.36.0 requires go >= 1.23.0 (running go 1.21.13; GOTOOLCHAIN=local)
```

so sad

atheo89 replied:
With that said, we should initiate golang upgrade to whole pipeline I guess?

jiridanek replied:
We can't, we don't have go-builder image for 1.23, latest is 1.22

jiridanek replied:
Opened Slack thread, hopefully somebody will have some comments. Until then, probably nothing to do here

https://redhat-internal.slack.com/archives/C05NXTEHLGY/p1741854502977309

jiridanek replied:
Conclusion is that we should be able to ignore this for now, until we can upgrade to go 1.23 easily. Should not be 2.19 release blocker.

harshad16 replied:
can we , we try go 1.22 , and see if the vulnerability can be fixed in this golang version ?
is there a reason to upgrade to 1.23 ?

jiridanek replied:
we can try go 1.22, but it won't help with x/net

go 1.22 is a good thing to try anyways, when there is a bit of slack time

dependabot[bot] committed:
fix(deps): bump golang.org/x/net in /components/notebook-controller

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.33.0 to 0.38.0.
- [Commits](https://github.com/golang/net/compare/v0.33.0...v0.38.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-version: 0.38.0
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump golang.org/x/net

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.33.0 to 0.38.0.
- [Commits](https://github.com/golang/net/compare/v0.33.0...v0.38.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-version: 0.38.0
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

jstourac commented:
/lgtm

If we merge this one, I presume we can then close the #589 and #590.

jiridanek commented:
> If we merge this one, I presume we can then close the https://github.com/opendatahub-io/kubeflow/pull/589 and https://github.com/opendatahub-io/kubeflow/pull/590.

Dependabot will autoclose them if we don't; i prefer to let dependabot to say all is updated ;p

/appove

jiridanek commented:
/approve
